### ISSUE TITLE
validation script doesn't link dependency libraries

### ISSUE THREAD

**Issue by suryaprakaz** [usnistgov/frvt#28]
Even if I place my 3rd party dependencies (*.so) under ./lib, the compile_and_link.sh script (invoked while running run_validate_11.sh) only takes the implementation library and I get a lot of undefined references. Am I missing something? Please help.

**Comment by mlngan** [comment]
You can use rpath=$ORIGIN, but you need to link your submission library against any 3rd party libraries that are required.  For example:

set (CMAKE_SHARED_LINKER_FLAGS "-Wl,-rpath,$ORIGIN")
add_library (frvt_11_null_001 SHARED nullimplfrvt11.cpp)
target_link_libraries(frvt_11_null_001 thirdpartylib1.so thirdpartylib2.so thirdpartylib3.so)

**Comment by avinashkshirsagar** [comment]
Hi suryaprakaz,
Were you able resolve your issue by linking third party libraries using the above code? Even after linking  I am getting Nof Found error (for ex. dlib not found) while executing the code.

**Comment by tanpengshi** [comment]
I have a model folder 'my_model' stored in ./lib 
But when I apply the codes above in the CMakeLists.txt in nullImpl folder

set (CMAKE_SHARED_LINKER_FLAGS "-Wl,-rpath,$ORIGIN")
add_library (frvt_11_null_001 SHARED nullimplfrvt11.cpp)
target_link_libraries(frvt_11_null_001 ${OpenCV_LIBS} ${PYTHON_LIBRARIES} my_model)

It gives an error while compiling with './scripts/build_null_impl.sh':
/usr/bin/ld: cannot find -lmy_model

Basically my embedded Python codes are calling the model directory to load into Tensorflow.

**Comment by xsacha** [comment]
This GitHub isn't intended for helping you learn Cmake :/
Unless libmy_model.so is a library, you don't need to add it to link libraries.
